### PR TITLE
Enable shared points for pair programming

### PR DESCRIPTION
**Pull request by delkopiso**
# What's up
When developers pair on a feature and one creates the pull request, only that developer gets all the points for the feature. In search of increased fairness in this regard there is a need to signal that a particular PR was the result of a combined effort by more than just the PR owner.

# What this does
- If a PR description has the phrase `paired with @abe` then the user with login `abe` gets to split the points on the PR with the creator.
- Also works when multiple users are listed e.g. `Paired with @abe, @edjohn` or `I paired with @abe and @edjohn`.

## Screenshots

- Merged a PR that was worth 100 points.
screenshot 2016-02-20 11 00 57

- Both the creator and the paired user received exactly 50 points.
screenshot 2016-02-20 11 01 13

### CODE REVIEW

**Comment by carlosepp** [comment]
Bring it on!!!!! :heart_eyes:

File: app/controllers/api/pull_requests_controller.rb [hunk 1]
```diff
@@ -1,7 +1,9 @@
 module Api
   class PullRequestsController < ApplicationController
     def create
-      PullRequest.create(data: pull_request_params) if action == 'closed'
+      if action == 'closed'
+        PointsMarshaller.new(data: pull_request_params).marshall
```
Comment (abepetrillo) [inline app/controllers/api/pull_requests_controller.rb]: is it `marshall` or `marshal`

File: app/models/points_marshaller.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  def initialize(data:)
+    @data = data
+  end
+
+  attr_accessor :data
+
+  def marshall
+    pull_request = PullRequest.create(data)
+    if pull_request.present? && pull_request.persisted?
```
Comment (abepetrillo) [inline app/models/points_marshaller.rb]: Can we not just do `if PullRequest.create(data)` ?

File: app/models/points_marshaller.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  def create_reviews(pr)
+    review_points = (pr.points / 2).round
+    pr.comments.each do |comment|
+      reviewer = User.where(nickname: comment.user.login).first
+      pull_request_review = PullRequestReview.create(
+        user: reviewer,
+        body: comment.body,
+        pull_request: pr
+      )
+      if pull_request_review.present? && pull_request_review.persisted?
```
Comment (abepetrillo) [inline app/models/points_marshaller.rb]: same here

File: app/models/points_marshaller.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        Score.give(time: pr.merged_at, user: reviewer, points: review_points)
+      end
+    end
+  end
+
+  def create_pairings(pr)
+    pair_points = (pr.points / pairers.size).round
+    pairers.each do |u|
+      pairing = Pairing.create(user: u, pull_request: pr)
+      if pairing.present? && pairing.persisted?
```
Comment (abepetrillo) [inline app/models/points_marshaller.rb]: and here

File: app/controllers/api/pull_requests_controller.rb (hunk 1 above)
Comment (delkopiso) [inline app/controllers/api/pull_requests_controller.rb]: oops

**Comment by delkopiso** [comment]
@abepetrillo fixed based on your comments

**Comment by abepetrillo** [comment]
:+1:

**Pull request merged**